### DISCUSSION TITLE
[Idea] Running 32-bit apps on newer macOS versions using Darling

### DISCUSSION THREAD

**Discussion by kristibektashi** [darlinghq/darling#1413] [Ideas]
(Disclaimer: I am not a maintainer in this repo, or in any way related to Darling or DarlingHQ, I just thought about this randomly)

Here's something that I thought about recently: Is it possible to port Darling to macOS, in order to run 32-bit x86 apps on versions that don't support them natively (Catalina and later)? This would be similar to what Wine and Crossover did in order to run 32-bit Windows apps in recent macOS versions, but for macOS apps instead.

This could start with 32-bit console app support, and once GUI support improves in Darling could also be extended to GUI apps. It could also potentially work on Apple Silicon Macs via Rosetta (Crossover proved that Rosetta can run 32-bit code)

Is this even plausible at all?

**Comment by CuriousTommy** [comment]
> Is it possible to port Darling to macOS.

Not in it's current form. One thing to keep in mind about Darling is that it relies heavily on Linux specific features. It would take a lot of work to get working on another OS (that isn't Linux).

As far as I am aware, the Darling team is not interested in getting Darling to work outside of Linux. Someone else would have to put in the effort to get Darling working outside of Linux.

**Reply by kristibektashi** [reply]
I understand what you are saying, however I am going to leave this discussion open in case someone is interested in this, as well as for general feedback on the idea. Who knows, perhaps a separate project will pop up that does exactly that.

**Comment by upintheairsheep** [comment]
There is a tool called 86x64 that can convert basic apps to 64 bit, but yes, I hate Catalina, it never added anything useful, just took away 32bit apps, Carbon, and Dashboard as well as splitting iTunes into a confusing mess.
